### ISSUE TITLE
Different behavior for accessibilityLabel in android/ios

### ISSUE THREAD

**Issue by jonathanmos** [facebook/react-native#32826]
### Description

On ios the accessibilityLabel defaults to a concatenation of the children's accessibilityLabels when undefined, whereas on android the contentDescription must be set explicitly with the accessibilityLabel prop. This leads to a situation where a View can have an accessibilityLabel set on ios but not on android. 

[The implementation on ios.](https://github.com/facebook/react-native/blob/3ba237b663215f66334bbe6f43a35a0a311dcf5c/React/Fabric/Mounting/ComponentViews/View/RCTViewComponentView.mm#L675) Here, ios takes the accessibilityLabels of all child views and appends them together if the parent does not have one explicitly set. E.g. if the parent has no accessibilityLabel, but childA has "hello" and childB has "world" then the accessibilityLabel of the parent becomes "hello world". On android the parent would have no accessibilityLabel in this situation. 

### Version

0.66.1

### Output of `npx react-native info`

System:
    OS: macOS 12.1
    CPU: (12) x64 Intel(R) Core(TM) i7-8850H CPU @ 2.60GHz
    Memory: 71.36 MB / 16.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 12.5.0 - /usr/local/opt/nvm/versions/node/v12.5.0/bin/node
    Yarn: 1.22.17 - /usr/local/opt/nvm/versions/node/v12.5.0/bin/yarn
    npm: 6.9.0 - /usr/local/opt/nvm/versions/node/v12.5.0/bin/npm
    Watchman: 2021.06.07.00 - /usr/local/bin/watchman
  Managers:
    CocoaPods: 1.10.1 - /usr/local/bin/pod
  SDKs:
    iOS SDK:
      Platforms: DriverKit 21.2, iOS 15.2, macOS 12.1, tvOS 15.2, watchOS 8.3
    Android SDK:
      Android NDK: 23.0.7344513-beta4
  IDEs:
    Android Studio: 4.1 AI-201.8743.12.41.7199119
    Xcode: 13.2.1/13C100 - /usr/bin/xcodebuild
  Languages:
    Java: 11.0.11 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 17.0.2 => 17.0.2
    react-native: 0.66.1 => 0.66.1
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found

### Steps to reproduce

Create a TouchableOpacity enclosing a Text element. Check the accessibilityLabel of the touchableOpacity - on android is will be empty, and on ios it will equal the value of the text element.

      <TouchableOpacity testID={'MY_NESTED_TEXT'}>
        <Text>test123</Text>
      </TouchableOpacity>

Run this test on android/ios:

    it('test accessibility label issue', () => {
      expect(element(by.id('MY_NESTED_TEXT'))).toHaveLabel('test123');
    });

**Comment by OfryL** [comment]
We found small workaround to make this work using `jsx-property-alias` plugin for babel: 
```json
{
  "plugins": [
    [
      "jsx-property-alias",
      {
        "properties": {
          "testID": "accessibilityLabel"
        }
      }
    ]
  ]
}
```
